server/sonar-server/src/main/java/org/sonar/server/computation/debt/CharacteristicImpl.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.computation.debt;
+
+import com.google.common.base.Preconditions;
+import org.apache.commons.lang.builder.ToStringBuilder;
+import org.apache.commons.lang.builder.ToStringStyle;
+
+public class CharacteristicImpl implements Characteristic {
+
+  private int id;
+  private String key;
+
+  public CharacteristicImpl(int id, String key) {
+    Preconditions.checkNotNull(key, "key cannot be null");
+    this.id = id;
+    this.key = key;
+  }
+
+  @Override
+  public int id() {
+    return id;
+  }
+
+  @Override
+  public String key() {
+    return key;
+  }
+
+  @Override
+  public String toString() {
+    return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE);
```

Comment:
for dumb bean like that, don't use reflection, that's stupid waste of CPU
there is convenient alternative (check the options provided by IDEA's toString generator, `Objects.toStringHelper` from Guava is nice enough but it will never beat a simple StringBuffer performance wise)

as a general practice, reflexion should never be used unless nothing can really be done otherwise
but in toString() more than anywhere else
one can not know how many times an Object's toString method may be called, directly or as part of wrapping objects
in addition, toString is one of these tricky cases where enabling log can actually make things worse in production